docs/resources/catalog.md

```diff
@@ -25,6 +25,7 @@ resource "databricks_catalog" "sandbox" {
 The following arguments are required:
 
 * `name` - Name of Catalog relative to parent metastore.
+* `metastore_id` - (Required if changing the metastore assigned to a workspace) If set, it must match the ID of the metastore assigned to the worspace. If not set, the current ID is exported.
```

Comment:
the statement `(Required if changing the metastore assigned to a workspace)` will likely cause confusion - let's just mention it in the [troubleshooting guide](docs/guides/troubleshooting.md)

Reply:
+1. IMO we don't need to show this to customers explicitly; the happy path would be not to specify the metastore ID, I think. Once these resources are available at the account-level, we'll change the docs to match what you've done for storage credentials.